src/test/java/rx/internal/operators/OperatorIgnoreElementsTest.java

```diff
@@ -127,4 +127,21 @@ public void onNext(Integer t) {
         assertEquals(0, count.get());
     }
     
+    @Test
+    public void testIgnoreElementsThen() {
+        final AtomicBoolean firstCompleted = new AtomicBoolean(false);
+        assertEquals(Arrays.asList(1, 2, 3), 
+                Observable
+                    .just("a","b","c")
+                    .doOnCompleted(new Action0() {
+
+                        @Override
+                        public void call() {
+                            firstCompleted.set(true);
+                        }})
+                    .ignoreElementsThen(Observable.just(1, 2, 3))
+                    .toList().toBlocking().single());
+        assertTrue(firstCompleted.get());
+    }
+    
```

Comment:
I'd also add a test to see if the source ignored emits an error and the other is not even subscribed. I'd do this in case one decides to write an operator directly for this use case instead of using concat.